src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -334,4 +296,117 @@ protected Object readProtocolWithCheckingBroken() {
     }
     return responses;
   }
+
+  private static class DefaultJedisSocketFactory implements JedisSocketFactory {
+
+    private String host;
+    private int port;
+    private int connectionTimeout;
+    private int soTimeout;
+    private boolean ssl;
+    private SSLSocketFactory sslSocketFactory;
+    private SSLParameters sslParameters;
+    private HostnameVerifier hostnameVerifier;
+
+    private DefaultJedisSocketFactory(String host, int port, int connectionTimeout, int soTimeout,
+        boolean ssl, SSLSocketFactory sslSocketFactory, SSLParameters sslParameters,
+        HostnameVerifier hostnameVerifier) {
+      this.host = host;
+      this.port = port;
+      this.connectionTimeout = connectionTimeout;
+      this.soTimeout = soTimeout;
+      this.ssl = ssl;
+      this.sslSocketFactory = sslSocketFactory;
+      this.sslParameters = sslParameters;
+      this.hostnameVerifier = hostnameVerifier;
+    }
+
+    @Override
+    public Socket createSocket() throws IOException {
+      Socket socket = null;
+      try {
+        socket = new Socket();
+        // ->@wjw_add
+        socket.setReuseAddress(true);
+        socket.setKeepAlive(true); // Will monitor the TCP connection is
+        // valid
+        socket.setTcpNoDelay(true); // Socket buffer Whetherclosed, to
+        // ensure timely delivery of data
+        socket.setSoLinger(true, 0); // Control calls close () method,
+        // the underlying socket is closed
+        // immediately
+        // <-@wjw_add
+
+        socket.connect(new InetSocketAddress(host, port), connectionTimeout);
```

Comment:
I this you should use the getHost(), getPort(), getConnectionTimeout()... so a user can override these methods

Reply:
Good point, will update.

Reply:
This is why I wanted DefaultJedisSocketFactory to be public, otherwise no-one can override it.

Reply:
Oh, I missed that, tell you what, I will just pull the class out and folks can do whatever logic they want.